File: Projeto-MobiFix/Database/dtos/stockDtos/encomendaStockDto.js
const paraEncomendaStockDto = (e) => {
    if (!e) return null;
    return {
        EncomendaID: e._id,
        PecaEAN: e.pecaId,
        Quantidade: e.quantidade,
        Estado: e.estado,
        DataPedido: e.dataPedido,
        OperadorRececaoNumero: e.operadorRececaoId || null,
        AdminValidadorNumero: e.adminValidadorId || null
    };
};

module.exports = { paraEncomendaStockDto };

File: Projeto-MobiFix/Backend/Models/Stocks/EncomendaStockDto.cs
namespace Backend.Models;

using System.ComponentModel.DataAnnotations;

public class EncomendaStockDto
{
    public int EncomendaID { get; set; }
    public string PecaEAN { get; set; } = string.Empty;
    public int Quantidade { get; set; }
    public string Estado { get; set; } = string.Empty;
    public string DataPedido { get; set; } = string.Empty;
    public string? OperadorRececaoNumero { get; set; }
    public string? AdminValidadorNumero { get; set; }
}

public class EncomendaStockCriacaoDto
{
    [Required]
    public string PecaEAN { get; set; } = string.Empty;
    [Required]
    [Range(1, int.MaxValue)]
    public int Quantidade { get; set; }
    public string? AdminValidadorNumero { get; set; }
}

public class EncomendaStockAtualizacaoDto
{
    [Required]
    public string Estado { get; set; } = string.Empty;
    public string? OperadorRececaoNumero { get; set; }
}
